src/CachePlugin.php

```diff
@@ -282,12 +285,13 @@ private function getCacheControlDirective(ResponseInterface $response, $name)
      */
     private function createCacheKey(RequestInterface $request)
     {
-        $body = (string) $request->getBody();
-        if (!empty($body)) {
-            $body = ' '.$body;
+        if (null === $this->config['cache_key_generator']) {
```

Comment:
I would move this check right after the option resolution. I don't see any benefit in timing here, this will be resolved sooner or later.